salt/minion.py

```diff
@@ -1591,7 +1592,7 @@ def _send_req_sync(self, load, timeout):
             )
             load["sig"] = sig
 
-        with salt.transport.client.ReqChannel.factory(self.opts) as channel:
+        with salt.channel.client.ReqChannel.factory(self.opts) as channel:
```

Comment:
We discussed this scale issue "in person" about a week ago, but I'll ask in the code review as well...  So here in the _send_req_sync() routine that is used to return job results from the minion (from a spawned thread or process), we are instantiating a brand new channel. When rabbitmq transport is used, this ends up being a very heavy operation (a new RMQ connection, RMQ channel, etc.), which presents a scale issue when 10K minions are returning job results. 

RabbitMQ connections are expensive and RabbitMQ prefers long-lasting connections at scale, which is what we already have with 3 out of 4 transport endpoints in master (1. PubServer, 2. ReqServer) and minion (3. PubClient). The 4. ReqClient is an on-demand short-lived channel, the way it is currently used.

Note: I tried caching the channel between different invocations of _send_req_sync() routine (e.g. between different minion jobs), but that does not work well between different threads OR processes that are spawned by the minion to run jobs (e.g. io_loop sharing issues, RMQ connection termination because io_loop was being blocked, etc.). 

The proposal was to use an IPC mechanism in the minion to communicate job results to an IPC server and have the IPC server use a persistent channel to the message broker.

A few questions about this approach: 
  - would this work for all transports? Is this a good change to have for all transports? 
  - would this scale when minion runs multiple concurrent jobs (is this common)?
  - salt.transport.client.PullChannel and salt.transport.client.PushChannel are a good fit for this?

Reply:
The master runs engines i.e. https://docs.saltproject.io/en/latest/topics/engines/index.html and https://github.com/saltstack/salt/blob/master/salt/engines/test.py the minion is not structure this way, need to consider Windows and other platforms.

Reply:
We will address this in a follow up PR. #61274